docbrown/core/src/tgraph.rs

```diff
@@ -1064,85 +1248,44 @@ mod graph_test {
 
         // the outbound neighbours of 9 at time 0..2 is the empty set
         let actual: Vec<u64> = g
-            .neighbours_window(9, &(0..2), Direction::OUT)
-            .map(|e| e.global_dst())
+            .edges_window(9, &(0..2), Direction::OUT)
+            .map(|e| e.dst_g_id)
             .collect();
         let expected: Vec<u64> = vec![];
         assert_eq!(actual, expected);
 
         // the outbound neighbours of 9 at time 0..4 are 1
         let actual: Vec<u64> = g
-            .neighbours_window(9, &(0..4), Direction::OUT)
-            .map(|e| e.global_dst())
+            .edges_window(9, &(0..4), Direction::OUT)
+            .map(|e| e.dst_g_id)
             .collect();
         assert_eq!(actual, vec![1]);
 
         // the inbound neighbours of 1 at time 0..4 are 9
         let actual: Vec<u64> = g
-            .neighbours_window(1, &(0..4), Direction::IN)
-            .map(|e| e.global_src())
+            .edges_window(1, &(0..4), Direction::IN)
+            .map(|e| e.src_g_id)
             .collect();
         assert_eq!(actual, vec![9]);
     }
 
-    #[test]
-    fn does_not_have_edge() {
-        let mut g = TemporalGraph::default();
-        g.add_vertex(1, 9);
-        g.add_vertex(2, 11);
-        g.add_edge(3, 11, 9);
-
-        let actual: bool = g.has_edge(9, 11);
-
-        assert_eq!(actual, false);
-    }
-
     #[test]
     fn has_edge() {
         let mut g = TemporalGraph::default();
+        g.add_vertex(1, 8);
         g.add_vertex(1, 9);
+        g.add_vertex(2, 10);
         g.add_vertex(2, 11);
+        g.add_edge(3, 9, 8);
+        g.add_edge(3, 8, 9);
         g.add_edge(3, 9, 11);
 
-        let actual: bool = g.has_edge(9, 11);
-
-        assert_eq!(actual, true);
-    }
-
-    #[test]
-    fn has_double_edge() {
-        let mut g = TemporalGraph::default();
-        g.add_vertex(1, 9);
-        g.add_vertex(2, 11);
-        g.add_edge(3, 9, 11);
-        g.add_edge(4, 11, 9);
-
-        let actual: bool = g.has_edge(9, 11);
-        let actual2: bool = g.has_edge(11, 9);
-
-        assert_eq!(actual, true);
-        assert_eq!(actual2, true);
-    }
-
-    #[test]
-    fn has_remote_out_edge() {
-        let mut g = TemporalGraph::default();
-        g.add_vertex(1, 9);
-        g.add_edge_remote_out(2, 9, 7, &vec![("bla".to_string(), Prop::U32(1))]);
-
-        let actual: bool = g.has_edge(9, 7);
-        assert_eq!(actual, true);
-    }
-
-    #[test]
-    fn has_remote_in_edge() {
-        let mut g = TemporalGraph::default();
-
-        g.add_vertex(1, 9);
-        g.add_edge_remote_into(2, 7, 9, &vec![("bla".to_string(), Prop::U32(1))]);
-
-        let actual: bool = g.has_edge(9, 7);
-        assert_eq!(actual, true);
+        assert_eq!(g.has_edge(8, 9), true);
+        assert_eq!(g.has_edge(9, 8), true);
+        assert_eq!(g.has_edge(9, 11), true);
+        assert_eq!(g.has_edge(11, 9), true);
```

Comment:
From the comment above, this should return false, the edge 11,9 was never added.